tensorflow_datasets/image/imagewang.py

```diff
@@ -0,0 +1,140 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Imagewang contains Imagenette and Imagewoof combined. """
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+# TODO(imagewang): BibTeX citation
+# See https://github.com/fastai/imagenette/issues/16.
+_CITATION = """
+"""
+
+_DESCRIPTION = """\
+Imagewang contains Imagenette and Imagewoof combined
+Image网 (pronounced "Imagewang"; 网 means "net" in Chinese) contains Imagenette
+and Imagewoof combined, but with some twists that make it into a tricky
+semi-supervised unbalanced classification problem:
+
+* The validation set is the same as Imagewoof (i.e. 30% of Imagewoof images);
+  there are no Imagenette images in the validation set (they're all in the
+  training set)
+* Only 10% of Imagewoof images are in the training set!
+* The remaining are in the unsup ("unsupervised") directory, and you can not
+  use their labels in training!
+* It's even hard to type and hard to say!
+
+The dataset comes in three variants:
+  * Full size
+  * 320 px
+  * 160 px
+This dataset consists of the Imagenette dataset {size} variant.
+"""
+
+_LABELS_FNAME = "image/imagewang_labels.txt"
+_URL_PREFIX = "https://s3.amazonaws.com/fast-ai-imageclas"
+_SIZES = ["full-size", "320px", "160px"]
+
+_SIZE_TO_DIRNAME = {
+    "full-size": "imagewang",
+    "320px": "imagewang-320",
+    "160px": "imagewang-160"
+}
+
+
+class ImagewangConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for Imagewang."""
+
+  def __init__(self, size, **kwargs):
+    super(ImagewangConfig, self).__init__(
+        version=tfds.core.Version("2.0.0"), **kwargs)
+    self.size = size
+
+
+def _make_builder_configs():
+  configs = []
+  for size in _SIZES:
+    configs.append(
+        ImagewangConfig(
+            name=size,
+            size=size,
+            description=_DESCRIPTION.format(size=size)))
+  return configs
+
+
+class Imagewang(tfds.core.GeneratorBasedBuilder):
+  """ Imagewang contains Imagenette and Imagewoof combined. """
+
+  VERSION = tfds.core.Version("2.0.0")
+  BUILDER_CONFIGS = _make_builder_configs()
+
+  def _info(self):
+    names_file = tfds.core.get_tfds_path(_LABELS_FNAME)
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(),
+            "label": tfds.features.ClassLabel(names_file=names_file)
+        }),
+        supervised_keys=("image", "label"),
+        homepage='https://dataset-homepage://github.com/fastai/imagenette',
```

Comment:
looks like an error in the URL

Reply:
This is actually the correct URL as all Imagenette, Imagewoof and Imagewang are all in the same repository.

Reply:
I meant having "://" in the middle of the url ;-)
Shouldn't it be just: "https://github.com/fastai/imagenette"

Reply:
😬